src/bundle/Resources/public/js/scripts/fieldType/base/base-field.js

```diff
@@ -0,0 +1,90 @@
+            [...doc.querySelectorAll(selector)].forEach(item => item.removeEventListener(eventName, callback, false));
+        }
+
+        findStateNodes(fieldNode, input, selectors) {
+            return selectors.reduce((total, selector) => total.concat([...fieldNode.querySelectorAll(selector)]), []);
+        }
+
+        toggleInvalidState(isError, config, input) {
+            const fieldNode = input.closest(this.fieldSelector);
+            const methodName = isError ? 'add' : 'remove';
+            const stateNodes = this.findStateNodes(fieldNode, input, config.invalidStateSelectors);
+
+            fieldNode.classList[methodName](this.classInvalid);
+            input.classList[methodName](this.classInvalid);
+
+            stateNodes.forEach(el => el.classList[methodName](this.classInvalid));
+        }
+
+        createErrorNode(message) {
+            const errorNode = doc.createElement('em');
+
+            errorNode.classList.add('ez-field-error');
+            errorNode.innerHTML = message;
+
+            return errorNode;
+        }
+
+        findErrorNodes(fieldNode, input, selectors) {
+            return selectors.reduce((total, selector) => total.concat([...fieldNode.querySelectorAll(selector)]), []);
+        }
+
+        findExistingErrorNodes(fieldNode, input, selectors) {
+            return this.findErrorNodes(fieldNode, input, selectors);
+        }
+
+        toggleErrorMessage(validationResult, config, input) {
+            const fieldNode = input.closest(this.fieldSelector);
+            const errorNodes = this.findErrorNodes(fieldNode, input, config.errorNodeSelectors);
+            const existingErrorSelectors = config.errorNodeSelectors.map(selector => selector + ' .ez-field-error');
+            const existingErrorNodes = this.findExistingErrorNodes(fieldNode, input, existingErrorSelectors);
+
+            existingErrorNodes.forEach(el => el.remove());
+
+            if (validationResult.isError) {
+                const errorNode = this.createErrorNode(validationResult.errorMessage);
+
+                errorNodes.forEach(el => el.append(errorNode));
+            }
+        }
+
+        validateField(config, event) {
+            const validationResult = this[config.callback](event);
+
+            if (!validationResult) {
+                return;
+            }
+
+            this.toggleInvalidState(validationResult.isError, config, event.target);
+            this.toggleErrorMessage(validationResult, config, event.target);
+        }
+
+        init() {
+            this.eventsMap.forEach(eventConfig => {
+                eventConfig.validateField = this.validateField.bind(this, eventConfig);
+
+                this.attachEvent(eventConfig.eventName, eventConfig.selector, eventConfig.validateField);
+            });
+        }
+
+        reinit() {
+            this.eventsMap.forEach(eventConfig => this.removeEvent(eventConfig.eventName, eventConfig.selector, eventConfig.validateField));
```

Comment:
These properties: `eventConfig.eventName, eventConfig.selector, eventConfig.validateField` could be extracted to a `const` for better readability. Just like this: `({eventName, selector, validateField}) => this.removeEvent(...)`.